File: cpp/remove_element.cpp
// https://leetcode.com/problems/remove-element/
// Easy

class Solution {
public:
    int removeElement(std::vector<int>& nums, int val) {
        int index = 0;
        int i = 0;
        while (i < nums.size()) {
            if (nums[i] != val) {
                nums[index] = nums[i]; 
                index++;
            }
            i++;
        }
        return index;
    }
};

File: python/remove_element.py
# https://leetcode.com/problems/remove-element/
# Easy


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        index = 0

        for i, _ in enumerate(nums):
            if nums[i] != val:
                nums[index] = nums[i]
                index += 1

        return index

